plugins/components/conversionlayer.go

```diff
@@ -397,39 +397,53 @@ func fillFlagMaps(c *Context, baseContext *cli.Context, originalFlags []Flag) er
 	// Loop over all plugin's known flags.
 	for _, flag := range originalFlags {
 		if stringFlag, ok := flag.(StringFlag); ok {
-			finalValue, err := getValueForStringFlag(stringFlag, baseContext.String(stringFlag.Name))
+			finalValue, skip, err := getValueForStringFlag(stringFlag, baseContext)
 			if err != nil {
 				return err
 			}
+			if skip {
+				continue
+			}
 			c.stringFlags[stringFlag.Name] = finalValue
-			continue
 		}
 
 		if boolFlag, ok := flag.(BoolFlag); ok {
-			c.boolFlags[boolFlag.Name] = getValueForBoolFlag(boolFlag, baseContext)
+			if baseContext.IsSet(boolFlag.Name) {
+				finalValue, skip := getValueForBoolFlag(boolFlag, baseContext)
+				if !skip {
+					c.boolFlags[boolFlag.Name] = finalValue
+				}
+			}
 		}
 	}
 	return nil
 }
 
-func getValueForStringFlag(f StringFlag, receivedValue string) (finalValue string, err error) {
-	if receivedValue != "" {
-		return receivedValue, nil
+func getValueForStringFlag(f StringFlag, baseContext *cli.Context) (finalValue string, skip bool, err error) {
+	value := baseContext.String(f.Name)
+	if value != "" {
+		finalValue = value
+		return
 	}
-	// Empty but has a default value defined.
 	if f.DefaultValue != "" {
-		return f.DefaultValue, nil
+		// Empty but has a default value defined.
+		finalValue = f.DefaultValue
+		return
 	}
+	skip = !baseContext.IsSet(f.Name)
 	// Empty but mandatory.
 	if f.Mandatory {
-		return "", errors.New("Mandatory flag '" + f.Name + "' is missing")
+		err = errors.New("Mandatory flag '" + f.Name + "' is missing")
 	}
-	return "", nil
+	return
 }
 
-func getValueForBoolFlag(f BoolFlag, baseContext *cli.Context) bool {
+func getValueForBoolFlag(f BoolFlag, baseContext *cli.Context) (finalValue, skip bool) {
 	if f.DefaultValue {
-		return baseContext.BoolT(f.Name)
+		finalValue = baseContext.BoolT(f.Name)
+		return
 	}
-	return baseContext.Bool(f.Name)
+	skip = !baseContext.IsSet(f.Name)
```

Comment:
I'm not sure we want to skip adding a bool flag with default value false, this is a legitimate value. Maybe we shouldn't skip bool flags at all.